<launch>

  <arg name="gui" default="true" />
  <arg name="device" default="0" />
  <arg name="video_device" default="/dev/video0" />

  <node name="usb_cam"
        pkg="usb_cam" type="usb_cam_node"
        output="screen" clear_params="true" >
    <rosparam subst_value="true" >
      video_device: $(arg video_device)
    </rosparam>
  </node>

  <node name="fg_node"
        pkg="fg_ros" type="fg_node.py"
        output="screen"
        clear_params="true"
        required="true">
    <remap from="~input" to="/usb_cam/image_raw" />
    <rosparam subst_value="true">
      model_path: $(find fg_ros)/trained_data/yolo7/2023-03-06-fg-for-dspl.pt
      score_thresh: 0.4
      nms_thresh: 0.45
      device: $(arg device)
      class_names: ['bamboo shoot', 'chocolate', 'coffee', 'coke', 'green tea', 'noodle', 'others']
    </rosparam>
  </node>

  <node name="label_image_decomposer_instance"
        pkg="jsk_perception" type="label_image_decomposer.py"
        output="screen" clear_params="true" >
    <remap from="~input" to="/fg_node/output" />
    <remap from="~input/label" to="/fg_node/output/label_ins" />
    <remap from="~output/label_viz" to="~output/viz" />
    <rosparam subst_value="true" >
      approximate_sync: false
      use_label_array: false
      only_label: false
      queue_size: 30
      alpha: 0.5
    </rosparam>
  </node>

  <node name="draw_rects"
        pkg="jsk_perception" type="draw_rects.py"
        output="screen"
        clear_params="true" >
    <remap from="~input" to="label_image_decomposer_instance/output/viz" />
    <remap from="~input/rects" to="fg_node/output/rects" />
    <remap from="~input/class" to="fg_node/output/class" />
    <rosparam subst_value="true">
      font_path: $(find jsk_recognition_utils)/font_data/NotoSansJP-Regular.otf
      use_classification_result: true
      label_size: 16
    </rosparam>
  </node>

  <group if="$(arg gui)" >
    <node name="image_view00"
          pkg="image_view" type="image_view">
      <remap from="image" to="draw_rects/output" />
    </node>
  </group>

</launch>
